config/dynamic.js

```diff
@@ -0,0 +1,85 @@
+const { merge } = require('webpack-merge');
+var base = require('./index.js')
+
+const apolloBatching = process.env.APOLLO_BATCH !== 'false';
+const memcachedServers = process.env.MEMCACHE_HOST || 'ui-memcached:11211';
+const baseUrl = process.env.BASE_URL || 'development.kiva.org';
+const env = process.env.SHORT_ENV || 'dev';
+const enableOptimizely = process.env.ENABLE_OPTIMIZELY !== 'false';
+const enablePerimeterx = process.env.ENABLE_PERIMETERX !== 'false';
+
+// Auth0 app IDs
+const adminAuthId = process.env.ADMIN_AUTH_ID || 'cNTV7eN5sBKgv9nQOxDpAz1pPfJGlBI5';
+const partnerAuthId = process.env.PARTNER_AUTH_ID || 'e6wSaTBDpKRkV5SV5cWw6zD6eJjd2DEk';
+const wwwAuthId = process.env.WWW_AUTH_ID || 'xOXldYg02WsLnlnn0D5xoPWI2i3aNsFD';
+const browserClientID = process.env.BROWSER_CLIENT_ID || 'ouGKxT4mE4wQEKqpfsHSE96c9rHXQqZF';
+const serverClientID = process.env.SERVER_CLIENT_ID || 'KIzjUBQjKZwMRgYSn6NvMxsUwNppwnLH';
+
+module.exports = merge(base, {
+	app: {
+		apolloBatching,
+		host: `www.${baseUrl}`,
+		publicPath: `https://www.${baseUrl}/`,
+		photoPath: process.env.PHOTO_PATH || 'https://www-dev-kiva-org.freetls.fastly.net/img/',
+		graphqlUri: `https://gateway.${baseUrl}/graphql`,
+		enablePerimeterx,
+		perimeterxAppId: process.env.PERIMETERX_APP_ID || 'PX5u4Lz98O',
+		enableAnalytics: true,
+		enableGTM: true,
+		googleTagmanagerId: process.env.GOOGLE_TAG_MANAGER_ID || 'GTM-PXFRMT',
+		enableGA: true,
+		gaId: process.env.GOOGLE_ANALYTICS_ID || 'UA-11686022-3',
+		grecaptchaSitekey: '6LcXENcmAAAAAEC4ygspn1WTm4zP4gLexXDnWuXE',
+		enableHotjar: true,
+		hotjarId: process.env.HOTJAR_ID || '3071239',
+		enableSnowplow: true,
+		snowplowUri: 'events.fivetran.com/snowplow/v5qt54ocr2nm',
+		enableFB: true,
+		enableOptimizely,
+		fbApplicationId: process.env.FACEBOOK_APP_ID || '364044572460',
+		fbOgNameSpace: `${env}-kiva`,
+		enableSentry: true,
+		sentryURI: 'https://7ce141b23c4a4e6091c206d08442f0e9@o7540.ingest.sentry.io/1201287',
+		algoliaConfig: {
+			group: `${env}`,
+			appId: 'testingZ9YK0WNQ85',
+			apiKey: '28aaa1d56e6a1688f13f6b41da0f27a5',
+			defaultIndex: `${env}_fundraising_popularity`,
+		},
+		btTokenKey: 'sandbox_q7kbtvzy_vynd473f79sfskz3',
+		auth0: {
+			loginRedirectUrls: {
+				[adminAuthId]: `https://admin.${baseUrl}/admin/login?force=1`,
+				[partnerAuthId]: `https://partners.${baseUrl}/pa2/login/login?authLevel=recent`,
+				[wwwAuthId]: `https://www.${baseUrl}/authenticate?authLevel=recent`,
+				[serverClientID]: `https://www.${baseUrl}/ui-login?force=true`,
+				[browserClientID]: `https://www.${baseUrl}/ui-login?force=true`,
+			},
+			enable: true,
+			checkFakeAuth: true,
```

Comment:
We would not want this to propagate to prod so it should be false by default and be setup to pass in as true only in lower envs...

Reply:
The way this is currently handled is that the key isn't defined in the `index.js` config file so it's false by default. Then the key is only added to lower env config files.

Reply:
Sorry, I'd mostly been setting this up mimicking the other lower envs, so it was importing from the prod (index.js) file and overriding that. I was picturing that we may do something similar to the current setup where prod is the default (perhaps create a duplicate of index.js for EKS configuration) that is then inherited from and overridden by the lower environments. Because of that, I hadn't really set this up for prod configuration yet. Do you think we should focus on using this `dynamic` config file for all EKS environments (and thus configure it with prod in mind) or continue with the existing configuration file inheritance pattern in this repo (and possibly spin up the base configuration file for EKS that contains prod values and is inherited by `dynamic`)?

Reply:
I am not sure if we want to tackle that problem in this PR (I would personally like it) but I think the end goal should definitely be allowing UI services to be configured entirely via secrets and environment variables, so a `dynamic` config seems attractive to me. I'll let @JoeTravisKiva chime in in terms of priority.

Reply:
This seemed to be trending toward something that might get used for all eks deployments so I was just trying to safe guard that incase it was fast tracked into use for prod...

If you want to skip the prod setup for now, I think we should name this file in a way that signifies it's only for lower envs, maybe `dynamic-dev.js`. It could also be good to startup a `dynamic-prod.js` for the prod base config which has a few values not present in the files that import it as a base.

Reply:
I think in terms of fast-tracking to prod, it would still just be EKS prod which is basically a pre-prod environment. We should still have time to vet it. Totally get what you're saying though. I'd personally prefer to have a global dynamic config (I like the 12-factor philosophy: apps should have no awareness of their environment in their code, or "[strict separation of config from code](https://12factor.net/config)").

Reply:
@mcstover and @hobbsh I hear you both. I've attempted to update the `dynamic` config to stand alone and work for prod. I pulled in all the properties that were only in the `index` file, so we no longer need to merge that in. At this point, pre-prod envs will have to set `CHECK_FAKE_AUTH` to true to enable that. Let me know if you have any thoughts on the current setup/default values.

Reply:
(Also, I alphabetized for ease of finding properties. I tried to put that in a separate commit for readability. Let me know if the previous grouping was preferable).